metaflow/includefile.py

```diff
@@ -173,7 +194,16 @@ def __init__(self, is_text, encoding, path):
 
     @classmethod
     def is_file_handled(cls, path):
+        # This returns a tuple:
+        #  - True/False indicating whether the file is handled
+        #  - None if we need to create a handler for the file, a LocalFile if
+        #    we already know what to do with the file or a Uploader if the file
+        #    is already present remotely (either starting with s3:// or local://)
+        #  - An error message if file is not handled
         if path:
+            if isinstance(path, IncludedFile):
```

Comment:
isn't this introducing a circular dependency?

Reply:
I am confused as to why this introduces a circular dependency. The class is defined on line 168 and returned by encode_url.

Reply:
And the `LocalFile` class is accessed in L:181 within `IncludedFile`